src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -997,6 +1000,143 @@ public static IEnumerable<object[]> VerifyUploadServersStreamsAndExpectedData
             }
         }
 
+        [Fact]
+        public async Task PostAsync_ResponseContentRead_RequestContentDisposedAfterResponseBuffered()
```

Comment:
> But it's not obvious when quickly looking at the test collateral that we have tests covering that.

I agree.  Part of the issue is that tests have been built up organically without much organization.  A good start I think will be splitting up HttpClientHandlerTest.cs.  At that time we can also revisit the naming and ordering of the tests.  That won't get us all the way there, but it'll make a solid dent I think.